### PR TITLE
Iter2 refactor

### PR DESCRIPTION
**Pull request by dwang733**
Fix css issues and some CodeClimate issues, and add some validations and tests for vendors and products.

### CODE REVIEW

**Comment by Juggernaut265** [comment]
All of the changes seem good to me. Good job on fixing the carousel_scripts code, as well as the divs.

File: app/assets/javascripts/carousel_scripts.js [hunk 1]
```diff
@@ -1,37 +1,32 @@
...
-		  element_index = $(this).index();
-		  if (element_index < hovered_index) {
-			$(this).context.classList.remove("spreadLeft"); 
-		  } 
-		  if (element_index > hovered_index) {
-			$(this).context.classList.remove("spreadRight"); 
-		  }
-	  	});
-  	});
+    helperFunction($(this), DOMTokenList.prototype.add);
```
Comment (ErikOstien) [inline app/assets/javascripts/carousel_scripts.js]: Deduplication looks good and formatting looks correct

File: features/edit_vendor.feature [hunk 2]
```diff
@@ -98,3 +105,11 @@ Feature: Edit an existing vendor in the database
     But I dismiss the popup
     Then I should be on the edit vendor page
     And I should see the vendor attributes filled in
+  
+  @javascript
+  Scenario: Edit vendor with a bad picture (sad)
+    When I include a bad picture
+    And I submit the form
+    Then I should see an error message on the edit vendor page
+    And I should see the vendor attributes, except "picture", filled in
```
Comment (ErikOstien) [inline features/edit_vendor.feature:114]: All new and edited cucumber tests look correct and they pass

**Review by ErikOstien — APPROVED** [review]
Lots of tab and spacing errors were fixed and looked correct. Does this branch solve all duplication issues? All other changes and refactoring look correct, and the checks and builds pass. Still maybe pushing the limit to the correct size of a pull request, but that can be attributed to all the tab/spacing corrections.

**Review by Juggernaut265 — COMMENTED** [review]
Thanks for making it clearer, in case we need to change anything going forward.

**Pull request merged**